hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -408,18 +365,32 @@ private boolean initializeFromFilesystem(String initializationTime, List<Metadat
       LOG.info("Initializing MDT partition " + partitionType + " at instant " + commitTimeForPartition);
 
       Pair<Integer, HoodieData<HoodieRecord>> fileGroupCountAndRecordsPair;
-      switch (partitionType) {
-        case FILES:
-          fileGroupCountAndRecordsPair = initializeFilesPartition(initializationTime, partitionInfoList);
-          break;
-        case BLOOM_FILTERS:
-          fileGroupCountAndRecordsPair = initializeBloomFiltersPartition(initializationTime, partitionToFilesMap);
-          break;
-        case COLUMN_STATS:
-          fileGroupCountAndRecordsPair = initializeColumnStatsPartition(partitionToFilesMap);
-          break;
-        default:
-          throw new HoodieMetadataException("Unsupported MDT partition type: " + partitionType);
+      try {
+        switch (partitionType) {
+          case FILES:
+            fileGroupCountAndRecordsPair = initializeFilesPartition(partitionInfoList);
+            // initialize the metadata reader again so the FILES partition can be read
+            initMetadataReader();
```

Comment:
This should be called in the end after setting table config because `HoodieBackedTableMetadata` checks for MDT init based on table config.